CONTRIBUTING.md

```diff
@@ -0,0 +1,62 @@
+# Contributing
+
+To make contributions to this charm, you'll need a working [development setup](https://juju.is/docs/sdk/dev-setup).
+
+You can create an environment for development with `tox`:
+
+```shell
+tox devenv -e integration
+source venv/bin/activate
+```
+
+## Testing
+
+This project uses `tox` for managing test environments. There are some pre-configured environments
+that can be used for linting and formatting code when you're preparing contributions to the charm:
+
+```shell
+tox -e fmt           # update your code according to linting rules
+tox -e lint          # code style
+tox -e static        # static type checking
+tox -e unit          # unit tests
+tox -e integration   # integration tests
+tox                  # runs 'fmt', 'lint', 'static', and 'unit' environments
+```
+
+## Build the charm
+
+Build the charm in this git repository using:
+
+```shell
+charmcraft pack
+```
+
+## Python versions
+
+The charm requires Python 3.12 to run/pack. If your system is using a different version of Python
+by default, you can do the following:
+```bash
+# install Python 3.12
+sudo add-apt-repository ppa:deadsnakes/ppa
+sudo apt update
+sudo apt install python3.12 python3.12-venv -y
+
+# Create venv with 3.12 and install tox. This is needed because tox will use the python
+# version of the Virtual Environment to create the final environment when we do "tox -e ..."
+python3.12 -m venv venv
+source dev-venv/bin/activate
+pip install tox
+```
+
+## Common Module
+
+The charm also hosts the `profile_management` package which aims to be abstracted to its own PyPi module in the
+future. The code for this should be able to be tested (unit/integration) in separate from the rest of the charm.
+
+For this reason we have developed the following poetry and tox environments:
+```bash
+# unit tests
+poetry install --with unit-library --no-root
```

Comment:
I pushed @kimwnasptd to keep everything in poetry because it makes more sense to have one tool handling all dependencies in a repo, rather than installing some through poetry and some through tox. FYI that usually [unit tests requirements are a superset of charm requirements](https://github.com/canonical/training-operator/blob/main/requirements-unit.in#L20), thus I think it is more intuitive if every dependency is handled with poetry rather than install some through poetry and some keep in tox.ini files

Reply:
I guess what I was asking is that in your `tox.ini` you can have:

```tox
[tests]
commands = 
    poetry install --with unit-library --no-root
    pytest <test file and args>
```

Does that collide with what you were saying @orfeas-k ?

Reply:
Ah I see what @DnPlas means. Yes, we shouldn't need to explicitly install the environment with `poetry` as it's already being installed with `tox -e unit-library` (in the tox's virtual env created when running the environment)

I pushed a commit to remove the extra command in the md file
291ba81